services/web/app.py

```diff
@@ -149,21 +157,21 @@ def __init__(self, *args, **kwargs):
             if "auth" in backend:
                 self.backends[backend["route"]] = backend["auth"]
 
-    def make_identity_header(self, identity_type, auth_data):
-        header_data = dict(type=identity_type, **auth_data)
-        print(header_data)
+    def make_identity_header(self, identity_type, auth_type, auth_data):
+        header_data = dict(type=identity_type, auth_type=auth_type, **{identity_type.lower(): auth_data})
+        app.logger.debug(header_data)
         return base64.encodebytes(json.dumps(header_data).encode("utf8")).replace(b"\n", b"")
 
     def auth_saml(self, auth):
         if "samlUserdata" in session:
             auth_data = session["samlUserdata"].items()
-            print(f"SAML auth_data: {auth_data}")
+            app.logger.debug(f"SAML auth_data: {auth_data}")
             predicate = auth["saml"]
             authorized = eval(predicate, dict(user=auth_data))
             if authorized:
                 resp = make_response("Authorized", 200)
                 resp.headers["X-RH-Identity"] = self.make_identity_header(
-                    "associate", dict(user={k: v if len(v) > 1 else v[0] for k, v in auth_data})
+                    "Associate", "saml-auth", {k: v if len(v) > 1 else v[0] for k, v in auth_data}
```

Comment:
The current identity header has a parent `identity` key, so I think we'll want it to look like:
```
{
  "identity": {
    "type": "Associate",
    "auth_type": "saml-auth",
    "associate": {
      "Role": [
        "role-1",
        "role-2"
      ],
      "email": "user@redhat.com",
      "givenName": "First",
      "rhatUUID": "abc-123",
      "surname": "Last"
    }
  }
}
```